### PR TITLE
Reference signing services via pulp_href (like everything else)

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jaiks@posteo.de>

fixes: https://pulp.plan.io/issues/9563

### CODE REVIEW

**Comment by jschmid1** [comment]
@quba42

**Comment by quba42** [comment]
@jschmid1 In order to link this fix with the issue, the Pulp CI will want:

- `closes #9563` in the commit message (not in the first line).
- A file named `CHANGES/9563.bugfix`. This file should contain a changelog entry like:
```
Fixed a bug requiring users to enter a full URL instead of just a ``pulp_href`` when creating an ``AptReleaseSigningService``.
```

**Comment by quba42** [comment]
And thanks for going ahead and just opening this PR, I know I was dragging my feet getting this done. :frowning_face:

**Comment by quba42** [comment]
@jschmid1 I have now verified/tested the fix (using the following version of the fix): https://github.com/pulp/pulp_deb/commit/f252699f8424451e0303947d86a0f7ccc9e0c387

You can either adjust this PR to match that version, or I can open a new PR and close this one. Either way works for me.

**Review by quba42 — CHANGES_REQUESTED** [review]
Some suggested and necessary changes, which I have also pushed to this repo here: https://github.com/pulp/pulp_deb/commit/f252699f8424451e0303947d86a0f7ccc9e0c387

File: pulp_deb/app/serializers/publication_serializers.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from rest_framework.serializers import BooleanField, ValidationError, HyperlinkedRelatedField
+from rest_framework.serializers import BooleanField, ValidationError, RelatedField
 from pulpcore.plugin.models import Publication
 from pulpcore.plugin.serializers import (
     DistributionSerializer,
```
[the remark this answers is not in the document]
Comment (ipanova) [reply pulp_deb/app/serializers/publication_serializers.py:4]: that's correct. you should import it from `pulpcore.plugin.serializers`

[the remark this answers is not in the document]
Comment (jschmid1) [reply pulp_deb/app/serializers/publication_serializers.py]: resolved in b0e94bf61d53ef98df363b5fcf310e1bb0baef2f

**Comment by jschmid1** [comment]
> Some suggested and necessary changes, which I have also pushed to this repo here: f252699

thanks, I rebased atop your suggestions.

**Comment by quba42** [comment]
I am afraid the reason this did not lint, is because I prematurely switched the issue tracker in the CI.
This should be fixed once https://github.com/pulp/pulp_deb/pull/425 is merged.

**Review by quba42 — APPROVED** [review]

**Comment by quba42** [comment]
@jschmid1 Can you rebase this on top of latest `main` branch one more time, now that I have merged #425, so the tests can run?

Once the tests turn green (they should) I will merge.

**Comment by quba42** [comment]
Thanks again @jschmid1.

I plan to do a release containing this change shortly (today or tomorrow).

**Pull request merged**